### ISSUE TITLE
Import packages, don't depend on them

### ISSUE THREAD

**Issue by hadley** [stan-dev/rstan#176]
Currently you have `Depends: Rcpp (>= 0.11.0), utils, inline, methods`, but these should all be moved into imports. It's bad practice (and not needed) to attach other packages to the search path when loading a package.

**Comment by bgoodri** [comment]
Thanks Hadley. 
- I'm not sure how `utils` got in there so it probably shouldn't be
- For `methods` the manual ( https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Namespaces-with-S4-classes-and-methods ) said to put it in Depends but then footnote 53 suggests maybe it is not necessary any more now that we require R >= 3.0.2.
- `Rcpp` is needed in Depends because if `Rcpp` is not loaded the user's model will fail with the message `Error in .Object$initialize(...) :  could not find function "cpp_object_initializer"` which is unexported by Rcpp
- I thought `inline` had a similar issue as `Rcpp` but now I think I'm wrong about that.

**Comment by hadley** [comment]
Are you using Rcpp modules? You shouldn't need to depend on Rcpp in most cases.

**Comment by bgoodri** [comment]
Yeah, it is this issue https://github.com/RcppCore/Rcpp/issues/168

**Comment by hadley** [comment]
I'll try and find some time to look into this in more detail. Would be nice if we could get a fix into Rcpp. I bet me and @wch can work it out between us.

**Comment by lionel-** [comment]
a temporary solution would be to import and reexport `cpp_object_initializer()`, this way Rcpp can be moved to `Imports`.

**Comment by bgoodri** [comment]
I've made this change in d531b6ee758991418dc5d6c03173fed44c97041c and it seems to work in every situation except compiling the vignette automatically when multiple cores are used, whereupon it says

```
Error: processing vignette 'rstan_vignette.Rnw' failed with diagnostics:
 chunk 5 (label = callstan) 
Error in unserialize(node$con) : error reading from connection
Execution halted
```

But that doesn't happen if cores = 1, or if I install the package without the vignettes and run a model manually with multiple cores, or even if I explicitly call tools::buildVignette(). Any ideas? I'm currently only using 1 core in the vignette.

**Comment by bgoodri** [comment]
This was largely fixed for rstan 2.9.0 or maybe 2.8.2, but I believe the "fix" entails that downstream packages of rstan that come with `stanmodel` objects have to put Rcpp in Depends. FYI, see the rstanarm package for how this can be done.
